### PR TITLE
Translate "ignored" closure expressions.

### LINKED ISSUE #24779: LLVM invalid linkage error from nested closures
Code:
```rust
(||||42)()()
```

Error:
```
Global is external, but doesn't have external or weak linkage!
i32 (%closure*)* @_ZN4main12closure.1067E
invalid linkage type for function declaration
i32 (%closure*)* @_ZN4main12closure.1067E
LLVM ERROR: Broken module found, compilation aborted!
```

### PR DESCRIPTION
**Pull request by eefriedman**
This isn't a very clean fix, but I'm not sure what a better fix would look
like.

Fixes #24779.

### CODE REVIEW

File: src/librustc_trans/trans/expr.rs [hunk 1]
```diff
@@ -139,9 +139,12 @@ pub fn trans_into<'blk, 'tcx>(bcx: Block<'blk, 'tcx>,
...
                 let lldest = PointerCast(bcx, lldest, val_ty(global));
                 memcpy_ty(bcx, lldest, global, expr_ty_adjusted(bcx, expr));
+                return bcx;
             }
-            // Don't do anything in the Ignore case, consts don't need drop.
-            return bcx;
+            // Even if we don't have a value to emit, and the expression
+            // doesn't have any side-effects, we still have to translate the
+            // body of any closures.
+            // TOOD: Find a better way of handling this case.
```
Comment (nrc) [inline src/librustc_trans/trans/expr.rs]: typo - TOOD - should be FIXME (TOOD will break make tidy)

**Comment by nrc** [comment]
r=me modulo the typo

**Comment by eefriedman** [comment]
Updated.

**Comment by alexcrichton** [comment]
Thanks! Could you squash the commits together as well?

**Comment by eefriedman** [comment]
Squashed.

**Pull request merged**